code/game/objects/structures/fullwindow.dm

```diff
@@ -58,6 +58,39 @@
 		icon_state = "[base_state][junction]"
 		return
 
+/obj/structure/window/full/verb/set_direction() //Full windows get this because it's possible for them to face diagonally
```

Comment:
I really do not support using verbs. They require a huge hodgepodge of boilerplate to get working correctly - for example, this is going to be ghost-exploitable.

I'd much rather you put spinning an unanchored window into the attack_hand for it and do it like that instead.